app/controllers/admin/orders_controller.rb

```diff
@@ -73,12 +99,40 @@ def load_order
 
   def update_order_status
     ActiveRecord::Base.transaction do
-      @order.tables.each do |table|
-        raise StandardError unless table.available!
+      raise StandardError unless @order.tables.each(&:available!)
 
-        OrderTable.where(order_id: @order.id).destroy_all
-        OrderDetail.where(order_id: @order.id).destroy_all if @order.cancel?
+      raise StandardError unless
+      OrderTable.where(order_id: @order.id).destroy_all
+
+      if @order.cancel?
+        @order.order_details.each(&:update_product_stock_after_cancel)
+        raise StandardError unless
+        OrderDetail.where(order_id: @order.id).destroy_all
       end
     end
   end
+
+  def not_accepted
+    return if @order.accepted?
+
+    flash[:danger] = t "order_not_accepted"
+    redirect_to admin_orders_path
+  end
+
+  def search_flash_success
+    flash.now[:success] =
+      I18n.t("search_with_result_order",
+             search_text: params[:search], count: @orders.size)
+  end
+
+  def order_update_table_transaction
+    raise StandardError unless @order.update(table_ids: @update_ids)
```

Comment:
cái này k cần raise. e chỉ cần update_attributes! là nó sẽ tự raise khi k update thành công

Reply:
vâng c, nhưng cái hàm update! kia e k biết nó raise ra error nào để rescue (e check thì toàn ra ctiveRecord::RecordInvalid), k biết có thêm cái nào khác k